conformance/lang/expressions/literals/string_literal.balt

```diff
@@ -0,0 +1,371 @@
+Labels: string
+
+public function main() {
+    string a = "\u{D800}"; // @error surrogate code points are not allowed
+    string b = "\u{D89F}"; // @error surrogate code points are not allowed
+    string c = "\u{DABF}"; // @error surrogate code points are not allowed
+    string d = "\u{DFFF}"; // @error surrogate code points are not allowed
+}
+
+Test-Case: output
+Description: Test Unicode noncharacters in string literal.
+Labels: string
+
+public function main() {
+    string a = "\u{FFFF}";
+    io:println(a); // @output ￿
+
+    string b = "\u{FFFE}";
+    io:println(b); // @output ￿
+}
+
+Test-Case: output
+Description: Test type of each constituent item in string value is string.
+Labels: member-access-expr, string
+
+public function main() {
+    string a = "constituent";
+    string b = a[0];
+    io:println(b); // @output c
+}
+
+Test-Case: output
+Description: Test empty string value.
+Labels: string, value:toBalString
+
+public function main() {
+    string a = "";
+    io:println(a.toBalString()); // @output ""
+}
+
+Test-Case: output
+Description: Test a sequence consisting of a single item v is the same thing as v. A single item is a sequence.
+Labels: string
+
+public function main() {
+    string a = "a";
+    io:println(a); // @output a
+}
+
+Test-Case: output
+Description: Test the type of the values in the iteration sequence of a string value is string.
+Labels: foreach-stmt, string
+
+public function main() {
+    string stringVar = "abc";
+    foreach string char in stringVar {
+        io:println(char); // @output a
+                          // @output b
+                          // @output c
+    }
+}
+
+Test-Case: output
+Description: Test completion value of string value iteration is nil.
+Labels: foreach-stmt, string, string:iterator, value:toBalString, while-stmt
+
+public function main() {
+    string stringVar = "a";
+    any char = "";
+    var iterator = stringVar.iterator();
+    while char != () {
+        char = iterator.next();
+        io:println(char.toBalString()); // @output {"value":"a"}
+                                        // @output ()
+    }
+
+    string stringVar2 = "";
+    io:println(stringVar2.iterator().next().toBalString()); // @output ()
+}
+
+Test-Case: error
+Description: Test the type of the values in the iteration sequence of a string value is string.
+Labels: foreach-stmt, string
+
+public function main() {
+    string stringVar = "abc";
+    foreach int val in stringVar { // @error values in the iteration sequence of a string value is string
+    }
+}
+
+Test-Case: output
+Description: Test concatination of strings that will result in a value looks like numeric escape.
+Labels: additive-expr, string
+
+public function main() {
+    string a = "\\" + "u{61}";
+    io:println(a); // @output \u{61}
+}
+
+Test-Case: error
+Description: Test type missmatch when assigning string literal to variables of type T, where string is not a sub type of T.
+Labels: string
```

Comment:
Don't we need to add other type-descriptors which are used in here?

Reply:
Will update thanx